gdpr/anonymizers/base.py

```diff
@@ -0,0 +1,169 @@
+        return self.get_decrypted_value(value)
+
+    def get_encryption_key(self):
+        """Return encryption key to use.
+
+        Override this method for custom logic.
+        """
+        if self._encryption_key is None:
+            raise ImproperlyConfigured()
+        return self._encryption_key
+
+    def get_anonymized_value(self, value: Any) -> Any:
+        """
+        There must be defined implementation of rule for anonymization.
+        To retain compatibility with older anonymizers this method is used as redirect to
+        `get_encrypted_value`.
+
+        Args:
+            value: value that is anonymize
+
+        Returns:
+            anonymized value
+        """
+        return self.get_encrypted_value(value)
+
+    def get_encrypted_value(self, value: Any) -> Any:
+        """
+        There must be defined implementation of rule for anonymization
+
+        Args:
+            value: value that is anonymize
+
+        Returns:
+            anonymized value
+        """
+        raise NotImplementedError
+
+    def get_decrypted_value(self, value: Any) -> Any:
+        """
+        There must be defined implementation of rule for anonymization
+
+        Args:
+            value: value that is anonymize
+
+        Returns:
+            anonymized value
+        """
+        if self.is_reversible:
+            raise NotImplementedError
+        return self.IrreversibleAnonymizationException()
+
+
+class NumericFieldAnonymizer(FieldAnonymizer):
+    max_anonymization_range: int
+
+    def __init__(self, max_anonymization_range: int = None, ignore_empty_values: bool = None,
+                 empty_values: Optional[List[Any]] = None):
+        if max_anonymization_range:
+            self.max_anonymization_range = max_anonymization_range
+        super().__init__(ignore_empty_values, empty_values)
+
+    def get_numeric_encryption_key(self, value: Union[int, float] = None) -> int:
+        if value is None:
+            return numerize_key(self.get_encryption_key()) % self.max_anonymization_range
+        # safety measure against key getting one bigger (overflow) on decrypt e.g. (5)=1 -> 5 + 8 = 13 -> (13)=2
+        guess_len = len(str(int(value)))
+        return numerize_key(self.get_encryption_key()) % 10 ** (guess_len if guess_len % 2 != 0 else (guess_len - 1))
+
+
+class ModelAnonymizerMeta(type):
+    """
+    Metaclass for anonymizers. The main purpose of the metaclass is to register anonymizers and find field anonymizers
+    defined in the class as attributes and store it to the fields property.
+    """
+
+    def __new__(cls, name, bases, attrs):
+        from gdpr.loading import anonymizer_register
+
+        new_obj = super().__new__(cls, name, bases, attrs)
+
+        # Also ensure initialization is only performed for subclasses of ModelAnonymizer
+        # (excluding Model class itself).
+        parents = [b for b in bases if isinstance(b, ModelAnonymizerMeta)]
```

Comment:
Proč tady není spíš `ModelAnonymizerBase`?

Reply:
převzato z předchozího kódu opravím

Reply:
Je to převzato z původního kódu a když tam dám `ModelAnonymizerBase` tak se všechno rozbije. Patřilo by tam  `ModelAnonymizerBase` kdyby to byla metoda `issubclass`. Třída definovaná sama o sobě není instance jejího rodiče.